aten/src/ATen/DLConvertor.cpp

```diff
@@ -388,4 +389,35 @@ Tensor fromDLPackVersioned(DLManagedTensorVersioned* src, std::function<void(voi
   return fromDLPackImpl<DLManagedTensorVersioned>(src, std::move(deleter));
 }
 
+Tensor maybeCopyTensor(
+    const Tensor& data,
+    std::optional<DLDevice> optional_dl_device,
+    std::optional<bool> copy) {
+  bool force_copy = copy.has_value() && *copy;
+  bool force_move = copy.has_value() && !*copy;
+
+  if (optional_dl_device.has_value()) {
+    auto device = at::getATenDevice(
+        optional_dl_device->device_type,
+        static_cast<c10::DeviceIndex>(optional_dl_device->device_id));
+
+    if (device != data.device()) {
+      TORCH_CHECK(
+          !force_move,
+          "cannot move tensor from ",
+          data.device(),
+          " to ",
+          device,
+          " without copying. Set copy=True is needed.");
```

Comment:
copy=None would also work. It's more that you cannot set copy=False in this case